hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -98,7 +99,7 @@
   private final transient HoodieMetrics metrics;
   private final transient HoodieCleanClient<T> cleanClient;
   private transient Timer.Context compactionTimer;
-
+  private transient WriteOperationType operationType;
```

Comment:
Can you move this AbstractHoodieWriteClient and use setters/getters in HoodieWriteClient